packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -310,10 +323,14 @@ export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapte
   }
   /** Dispose of the current CallWithChatAdapter. */
   public dispose(): void {
-    this.chatAdapter.offStateChange(this.onChatStateChange);
+    this.chatAdapterPromise.then((adapter) => {
+      adapter.offStateChange(this.onChatStateChange);
+    });
     this.callAdapter.offStateChange(this.onCallStateChange);
 
-    this.chatAdapter.dispose();
+    this.chatAdapterPromise.then((adapter) => {
```

Comment:
This is technically a breaking change. Because the chatAdapter.dispose is now wrapped in a promise, it doesn't execute sequentially anymore meaning it hasn't disposed when this method completes and the develpoer has no way to await this

Now that said, I can't think of any tangible, real impact of this to developers so maybe this instance it OK?

Reply:
Being discussed in another comment